pre_commit_hooks/check_vcs_permalinks.py

```diff
@@ -25,11 +24,23 @@ def _check_filename(filename: str) -> int:
 def main(argv: Optional[Sequence[str]] = None) -> int:
     parser = argparse.ArgumentParser()
     parser.add_argument('filenames', nargs='*')
+    parser.add_argument(
+        '--github-enterprise-domain', dest='github_enterprise_domain',
+    )
     args = parser.parse_args(argv)
 
     retv = 0
+
+    github_non_permalink = re.compile(NON_PERMALINK % b'github.com')
     for filename in args.filenames:
-        retv |= _check_filename(filename)
+        retv |= _check_filename(filename, github_non_permalink)
+
+    if args.github_enterprise_domain:
+        github_enterprise_non_permalink = re.compile(
+            NON_PERMALINK % args.github_enterprise_domain.encode(),
+        )
+        for filename in args.filenames:
+            retv |= _check_filename(filename, github_enterprise_non_permalink)
```

Comment:
this is going to re-parse twice, maybe add an additional argument to `_check_filename` such that it only runs over each file twice?  this will also help with output ordering as well

Reply:
Rather than increasing the logic branch, using a list can make it simpler, so I tried using a list. I tried modifying action='append' to allow users to add more than one domain. How about this?